services/sinner/contracts/destination/parser.go

```diff
@@ -0,0 +1,102 @@
+// Package destination is the execution hub contract parser.
+package destination
+
+import (
+	"context"
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/synapsecns/sanguine/agents/contracts/destination"
+	"github.com/synapsecns/sanguine/services/sinner/db"
+	"github.com/synapsecns/sanguine/services/sinner/db/model"
+	"github.com/synapsecns/sanguine/services/sinner/logger"
+	"github.com/synapsecns/sanguine/services/sinner/types"
+)
+
+// ParserImpl is the parser for the origin contract.
+type ParserImpl struct {
+	filterer *destination.DestinationFilterer
+	// parser is the parser interface.
+	parser destination.Parser
+	// db is the database
+	db db.EventDB
+	// chainID is the chain ID
+	chainID uint32
+	// txMap is a map of tx hashes to tx data, exported for testing.
+	TxMap map[string]types.TxSupplementalInfo
```

Comment:
I see why you want to export this, but what would be better is doing `txMap` here and adding a `UnsafeGetTXMap()` func and commenting it's for test only. That is if you even need to use it outside of this module